lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -123,41 +130,76 @@ public function convertException($message, DriverException $exception)
 
     /**
      * {@inheritdoc}
+     *
+     * @return AbstractPlatform|MariaDb102Platform|MySQL57Platform|MySqlPlatform
+     * @throws DBALException
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
+        if (false !== stripos($version, 'mariadb')
+            && version_compare($this->getMariaDbMysqlVersionNumber($version), '10.2.7', '>=')) {
+            return new MariaDb102Platform();
         }
 
-        if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
+        if (false === stripos($version, 'mariadb')
```

Comment:
The duplication of `if (stripos($version, 'mariadb'))` could be avoided:
```php
if ('mariadb') {
    if (version_compare()) return;
} else {
    if (version_compare()) return;
}
```

Reply:
Done